core/src/main/java/com/linecorp/armeria/server/healthcheck/DefaultCpuHealthChecker.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2015 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.healthcheck;
+
+import static java.util.Objects.requireNonNull;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+
+/**
+ * Forked from <a href="https://github.com/micrometer-metrics/micrometer/blob/8339d57bef8689beb8d7a18b429a166f6595f2af/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/system/ProcessorMetrics.java">ProcessorMetrics.java</a> in the micrometer core.
+ */
+public class DefaultCpuHealthChecker implements HealthChecker {
```

Comment:
If we hide the implementation from the public API, we can get more freedom on changes.
Should we add a factory method of this class to `HealthChecker` and make this package-private?
```
// The simplest one.
HealthChecker.ofCpu(cpuUsage);
// A complex configuation.
HealthChecker.ofCpu(cpuUsage, cpuIdle...);
```